libs/sdk-core/src/breez_services.rs

```diff
@@ -352,10 +357,10 @@ impl BreezServices {
         filter: PaymentTypeFilter,
         from_timestamp: Option<i64>,
         to_timestamp: Option<i64>,
-    ) -> Result<Vec<Payment>> {
+    ) -> SdkResult<Vec<Payment>> {
         self.persister
             .list_payments(filter, from_timestamp, to_timestamp)
-            .map_err(|err| anyhow!(err))
+            .map_err(|_| NewSdkError::SdkFailedToListPayments)
```

Comment:
Perhaps this pattern of mapping error from the storage layer to a specific error can be improved.
IMO we don't need specific error for every persistence failure (e.g FailedToListPayments, FailedToGetNodeState, etc...).
We can do the following:
1. Define one error for persistency: `PersistenceFailure`
2. Return this error from the storage layer directly which eliminate the need to map here at the BreezServices layer.